type assert with _ status

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

func main() {
        var i interface{} = 1
        var s struct{}
        s, _ = i.(struct{})
        fmt.Println(s)
}
```

Expected result:
```console
$ go run ./sample.go
{}
```

Got:
```console
$ yaegi ./sample.go
./sample.go:6:8: panic
runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000250600, 0x40)
        /home/marc/sdk/go1.13.12/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0001b5db0)
        /home/marc/src/yaegi/interp/interp.go:333 +0xbb
panic(0xd8cbe0, 0x16147d0)
        /home/marc/sdk/go1.13.12/src/runtime/panic.go:679 +0x1b2
github.com/containous/yaegi/interp.runCfg.func1(0xc0003000a0, 0xc000308f00)
        /home/marc/src/yaegi/interp/run.go:116 +0x20c
panic(0xd8cbe0, 0x16147d0)
        /home/marc/sdk/go1.13.12/src/runtime/panic.go:679 +0x1b2
github.com/containous/yaegi/interp.canAssertTypes(0x107b1e0, 0xd3c100, 0x0, 0x0, 0xd3c100)
        /home/marc/src/yaegi/interp/run.go:290 +0x34
github.com/containous/yaegi/interp.typeAssert2.func4(0xc0003000a0, 0xc0001847c0)
        /home/marc/src/yaegi/interp/run.go:276 +0x1cb
github.com/containous/yaegi/interp.runCfg(0xc000308f00, 0xc0003000a0)
        /home/marc/src/yaegi/interp/run.go:122 +0x7a
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000142300, 0xc000308500, 0xc000300000)
        /home/marc/src/yaegi/interp/run.go:100 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000142300, 0xc000243280, 0x78, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/src/yaegi/interp/interp.go:409 +0x53c
main.main()
        /home/marc/src/yaegi/cmd/yaegi/yaegi.go:166 +0x6e2
```

--- a/_test/interface45.go
+++ b/_test/interface45.go
@@ -1,0 +1,13 @@
+package main
+
+import "fmt"
+
+func main() {
+	var i interface{} = 1
+	var s struct{}
+	s, _ = i.(struct{})
+	fmt.Println(s)
+}
+
+// Output:
+// {}

--- a/interp/run.go
+++ b/interp/run.go
@@ -230,10 +230,11 @@ func typeAssert(n *node) {
 
 func typeAssert2(n *node) {
 	c0, c1 := n.child[0], n.child[1]
-	value := genValue(c0)              // input value
-	value0 := genValue(n.anc.child[0]) // returned result
-	value1 := genValue(n.anc.child[1]) // returned status
-	typ := c1.typ                      // type to assert or convert to
+	value := genValue(c0)                    // input value
+	value0 := genValue(n.anc.child[0])       // returned result
+	value1 := genValue(n.anc.child[1])       // returned status
+	setStatus := n.anc.child[1].ident != "_" // do not assign status to "_"
+	typ := c1.typ                            // type to assert or convert to
 	typID := typ.id()
 	rtype := typ.rtype // type to assert
 	next := getExec(n.tnext)
@@ -247,7 +248,9 @@ func typeAssert2(n *node) {
 			} else {
 				ok = false
 			}
-			value1(f).SetBool(ok)
+			if setStatus {
+				value1(f).SetBool(ok)
+			}
 			return next
 		}
 	case isInterface(typ):
@@ -257,7 +260,9 @@ func typeAssert2(n *node) {
 			if ok {
 				value0(f).Set(v)
 			}
-			value1(f).SetBool(ok)
+			if setStatus {
+				value1(f).SetBool(ok)
+			}
 			return next
 		}
 	case n.child[0].typ.cat == valueT:
@@ -267,7 +272,9 @@ func typeAssert2(n *node) {
 			if ok {
 				value0(f).Set(v)
 			}
-			value1(f).SetBool(ok)
+			if setStatus {
+				value1(f).SetBool(ok)
+			}
 			return next
 		}
 	default:
@@ -277,13 +284,18 @@ func typeAssert2(n *node) {
 			if ok {
 				value0(f).Set(v.value)
 			}
-			value1(f).SetBool(ok)
+			if setStatus {
+				value1(f).SetBool(ok)
+			}
 			return next
 		}
 	}
 }
 
 func canAssertTypes(src, dest reflect.Type) bool {
+	if dest == nil {
+		return false
+	}
 	if src == dest {
 		return true
 	}

